pkg/render/apiserver.go

```diff
@@ -542,12 +542,14 @@ func (c *apiServerComponent) apiServer() *appsv1.Deployment {
 	var replicas int32 = 1
 
 	hostNetwork := false
+	dnsPolicy := corev1.DNSClusterFirst
 	if c.installation.Spec.KubernetesProvider == operator.ProviderEKS &&
 		c.installation.Spec.CNI.Type == operator.PluginCalico {
 		// Workaround the fact that webhooks don't work for non-host-networked pods
 		// when in this networking mode on EKS, because the control plane nodes don't run
 		// Calico.
 		hostNetwork = true
+		dnsPolicy = corev1.DNSClusterFirstWithHostNet
```

Comment:
Should we include `Otherwise they fail to lookup services within the cluster` as a comment here?